<launch>
	<arg name="communication_delay_compensator_param_path"
			 default="$(find-pkg-share communication_delay_compensator)/param/communication_delay_compensator.param.yaml"/>
	<!--    <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory"/>-->
	<!--    <arg name="input/control_raw" default="/control/command/control_cmd"/>-->
	<!--    <arg name="input/measured_steering" default="/vehicle/status/steering_status"/>-->
	<!--    <arg name="input/current_odometry" default="/localization/kinematic_state"/>-->
	<!--    <arg name="output/error_stamped" default="/control_performance/performance_vars"/>-->
	<!--    <arg name="output/driving_status_stamped" default="/control_performance/driving_status"/>-->

	<!-- vehicle info -->
	<arg name="vehicle_info_param_file" default="$(find-pkg-share vehicle_info_util)/config/vehicle_info.param.yaml"/>

	<node pkg="communication_delay_compensator" exec="communication_delay_compensator_exe"
				name="communication_delay_compensator_exe" output="screen">
		<param from="$(var communication_delay_compensator_param_path)"/>
		<!--        <param from="$(var vehicle_info_param_file)"/>-->

		<remap from="~/input/control_cmd" to="/control/trajectory_follower/control_cmd"/>
		<remap from="~/input/steering_state" to="/vehicle/status/steering_status"/>
		<remap from="~/input/current_odometry" to="/localization/kinematic_state"/>

		<remap from="~/input/long_errors" to="/nmpc_error_report"/>
		<remap from="~/input/lat_errors" to="/nmpc_error_report"/>

		<!--        <remap from="~/input/cp_errors" to="/control_performance/performance_vars"/>-->
		<remap from="~/output/communication_delay_compensation_refs" to="observers/time_delay_compensator/delay_refs"/>
		<remap from="~/output/communication_delay_compensation_debug"
					 to="observers/time_delay_compensator/delay_debug"/>

		<!--        <remap from="~/input/reference_trajectory" to="$(var input/reference_trajectory)"/>-->
		<!--        <remap from="~/input/control_raw" to="$(var input/control_raw)"/>-->
		<!--        <remap from="~/input/measured_steering" to="$(var input/measured_steering)"/>-->
		<!--        <remap from="~/input/odometry" to="$(var input/current_odometry)"/>-->
		<!--        <remap from="~/output/error_stamped" to="$(var output/error_stamped)"/>-->
		<!--        <remap from="~/output/driving_status_stamped" to="$(var output/driving_status_stamped)"/>-->
	</node>
</launch>